java/src/main/java/com/microsoft/onnxruntime/ONNXJavaType.java

```diff
@@ -0,0 +1,118 @@
+    INT32(5, int.class, 4),
+    INT64(6, long.class, 8),
+    BOOL(7, boolean.class, 1),
+    STRING(8, String.class, 4),
+    UNKNOWN(0, Object.class, 0);
+
+    private static final ONNXJavaType[] values = new ONNXJavaType[9];
+
+    static {
+        for (ONNXJavaType ot : ONNXJavaType.values()) {
+            values[ot.value] = ot;
+        }
+    }
+
+    public final int value;
+    public final Class<?> clazz;
+    public final int size;
+
+    ONNXJavaType(int value, Class<?> clazz, int size) {
+        this.value = value;
+        this.clazz = clazz;
+        this.size = size;
+    }
+
+    /**
+     * Maps from an int in native land into an ONNXJavaType instance.
+     * @param value The value to lookup.
+     * @return The enum instance.
+     */
+    public static ONNXJavaType mapFromInt(int value) {
+        if ((value > 0) && (value < values.length)) {
+            return values[value];
+        } else {
+            return UNKNOWN;
+        }
+    }
+
+    /**
+     * Must match the values from ONNXUtil.c.
+     *
+     * @param onnxValue The value from ONNXUtil.c
+     * @return A JavaDataType instance representing the Java type
+     */
+    public static ONNXJavaType mapFromONNXTensorType(ONNXTensorType onnxValue) {
+        switch (onnxValue) {
+            case ONNX_TENSOR_ELEMENT_DATA_TYPE_UINT8:
+            case ONNX_TENSOR_ELEMENT_DATA_TYPE_INT8:
+                return ONNXJavaType.INT8;
+            case ONNX_TENSOR_ELEMENT_DATA_TYPE_UINT16:
+            case ONNX_TENSOR_ELEMENT_DATA_TYPE_INT16:
+                return ONNXJavaType.INT16;
+            case ONNX_TENSOR_ELEMENT_DATA_TYPE_UINT32:
+            case ONNX_TENSOR_ELEMENT_DATA_TYPE_INT32:
+                return ONNXJavaType.INT32;
+            case ONNX_TENSOR_ELEMENT_DATA_TYPE_UINT64:
+            case ONNX_TENSOR_ELEMENT_DATA_TYPE_INT64:
+                return ONNXJavaType.INT64;
+            case ONNX_TENSOR_ELEMENT_DATA_TYPE_FLOAT16:
+            case ONNX_TENSOR_ELEMENT_DATA_TYPE_FLOAT:
+                return ONNXJavaType.FLOAT;
+            case ONNX_TENSOR_ELEMENT_DATA_TYPE_DOUBLE:
+                return ONNXJavaType.DOUBLE;
+            case ONNX_TENSOR_ELEMENT_DATA_TYPE_STRING:
+                return ONNXJavaType.STRING;
+            case ONNX_TENSOR_ELEMENT_DATA_TYPE_BOOL:
+                return ONNXJavaType.BOOL;
+            case ONNX_TENSOR_ELEMENT_DATA_TYPE_UNDEFINED:
+            case ONNX_TENSOR_ELEMENT_DATA_TYPE_COMPLEX64:
+            case ONNX_TENSOR_ELEMENT_DATA_TYPE_COMPLEX128:
+            case ONNX_TENSOR_ELEMENT_DATA_TYPE_BFLOAT16:
+            default:
+                return ONNXJavaType.UNKNOWN;
+        }
+    }
+
+    /**
+     * Maps from a Java class object into the enum type, returning {@link ONNXJavaType#UNKNOWN}
+     * for unsupported types.
+     * @param clazz The class to use.
+     * @return An ONNXJavaType instance.
+     */
+    public static ONNXJavaType mapFromClass(Class<?> clazz) {
```

Comment:
>mapFromClas [](start = 31, length = 11)

Seems incomplete?

Reply:
I didn't think those types were supported in the runtime. Otherwise I can map them to boxed types for the moment. With value types there will be a Java answer but that's not available yet.